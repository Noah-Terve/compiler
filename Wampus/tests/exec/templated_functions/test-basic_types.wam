template @l T, hello @r
T test(T x, hello y) {
    T w = x;
    hello z = y;
    return x;
}

print @l int @r(test @l int, int @r (5, 4));
print @l int @r(test @l int, int @r (4, 5));
print @l int @r(test @l int, int @r (3, 6));
print @l int @r(test @l int, int @r (2, 7));
print @l int @r(test @l int, bool @r (5, true));
print @l int @r(test @l int, char @r (3, 'a'));
print @l int @r(test @l int, float @r (2, 7.91));
print @l int @r(test @l int, string @r (8, "not"));

print @l bool @r(test @l bool, int @r (true, 4));
print @l bool @r(test @l bool, bool @r (true, false));
print @l bool @r(test @l bool, char @r (true, 'a'));
print @l bool @r(test @l bool, float @r (false, 1.23));
print @l bool @r(test @l bool, string @r (false, "false"));

print @l char @r(test @l char, int @r ('a', 4));
print @l char @r(test @l char, bool @r ('w', true));
print @l char @r(test @l char, char @r ('a', 'b'));
print @l char @r(test @l char, float @r ('q', 3.14));
print @l char @r(test @l char, string @r ('b', "bee"));

print @l float @r(test @l float, int @r (1.2, 4));
print @l float @r(test @l float, bool @r (6.78, true));
print @l float @r(test @l float, char @r (123.456, 'y'));
print @l float @r(test @l float, float @r (1.0, 1.0));
print @l float @r(test @l float, string @r (08234.87, "maybe yes"));

print @l string @r(test @l string, int @r ("hello", 2));
print @l string @r(test @l string, bool @r ("hello", false));
print @l string @r(test @l string, char @r ("world", '!'));
print @l string @r(test @l string, float @r ("alive", 456.73));
print @l string @r(test @l string, string @r ("work", "please"));
