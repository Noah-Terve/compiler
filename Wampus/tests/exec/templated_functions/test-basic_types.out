5
4
3
2
5
3
2
8
1
1
1
0
0
a
w
a
q
b
1.2
6.78
123.456
1
8234.87
hello
hello
world
alive
work
